game/menu/pausemenu/pausemenu.gd

```diff
@@ -0,0 +1,70 @@
+extends CanvasLayer
+
+signal pause_menu_opened
+signal pause_menu_closed
+
+onready var _main := $Margin
+onready var _reset_btn := $Margin/VBox/restartbtn
+
+
+func _ready():
+	hide()
+
+
+func _input(event):
```

Comment:
If a "minigame" has its own menu, this could cause conflicts.
Maybe make a check if the game wants to handle this itself.

Reply:
I don't think a minigame should be allowed to habe its own menu

Reply:
_input() is the way to avoid conflicts.
If any menu further down the scene tree uses the event it won't reach the pause menu.
If the pause menu also doesn't use the event the other nodes can get it with [_unhandled_input()](https://docs.godotengine.org/en/stable/classes/class_node.html#class-node-method-unhandled-input)

Reply:
I did some tests and you're absolutely right.